frontend/pages/admin/IntegrationsPage/cards/CertificateAuthorities/components/AddCertAuthorityModal/AddCertAuthorityModal.tsx

```diff
@@ -64,34 +122,74 @@ const AddCertAuthorityModal = ({ onExit }: IAddCertAuthorityModalProps) => {
       onExit();
       setConfig(newConfig);
     } catch (e) {
-      renderFlash("error", generateErrorMessage(e));
+      renderFlash("error", getErrorMessage(e));
     }
     setIsAdding(false);
   };
 
+  const dropdownOptions = useMemo(() => {
+    return generateDropdownOptions(!!config?.integrations.ndes_scep_proxy);
+  }, [config?.integrations.ndes_scep_proxy]);
+
+  const renderForm = () => {
+    const submitBtnText = "Add CA";
+
+    switch (certAuthorityType) {
+      case "digicert":
+        return (
+          <DigicertForm
+            formData={digicertFormData}
+            submitBtnText={submitBtnText}
+            isSubmitting={isAdding}
+            onChange={onChangeForm}
+            onSubmit={onAddCertAuthority}
+            onCancel={onExit}
+          />
+        );
+      case "ndes":
+        return (
+          <NDESForm
+            formData={ndesFormData}
+            submitBtnText={submitBtnText}
+            isSubmitting={isAdding}
+            onChange={onChangeForm}
+            onSubmit={onAddCertAuthority}
+            onCancel={onExit}
+          />
+        );
+      case "custom":
+        return (
+          <CustomSCEPForm
+            formData={customSCEPFormData}
+            submitBtnText={submitBtnText}
+            isSubmitting={isAdding}
+            onChange={onChangeForm}
+            onSubmit={onAddCertAuthority}
+            onCancel={onExit}
+          />
+        );
+      default:
+        return null;
+    }
+  };
+
   return (
     <Modal
       className={baseClass}
       title="Add certificate authority (CA)"
       width="large"
       onExit={onExit}
+      isContentDisabled={isAdding}
```

Comment:
My only issue with using this built in `<Modal/>` param is that this does not actually disable the form's keyboard access. I shy away from using it because of this.